src/main/java/app/controller/FileParser.java

```diff
@@ -0,0 +1,119 @@
+                        break;
+                    case "wall":
+                        map.addWall(new Rectangle2D.Double(Double.parseDouble(coords[0]), Double.parseDouble(coords[1]), Integer.parseInt(coords[2]) - Integer.parseInt(coords[0]), Integer.parseInt(coords[3]) - Integer.parseInt(coords[1])));
+                        break;
+                    case "shaded":
+                        map.addShade(new Rectangle2D.Double(Double.parseDouble(coords[0]), Double.parseDouble(coords[1]), Integer.parseInt(coords[2]) - Integer.parseInt(coords[0]), Integer.parseInt(coords[3]) - Integer.parseInt(coords[1])));
+                        break;
+                    case "tower":
+                        map.addTower(new Rectangle2D.Double(Double.parseDouble(coords[0]), Double.parseDouble(coords[1]), Integer.parseInt(coords[2]) - Integer.parseInt(coords[0]), Integer.parseInt(coords[3]) - Integer.parseInt(coords[1])));
+                        break;
+                    case "teleport":
+                        map.addPortal(new Rectangle2D.Double(Double.parseDouble(coords[0]), Double.parseDouble(coords[1]), Integer.parseInt(coords[2]) - Integer.parseInt(coords[0]), Integer.parseInt(coords[3]) - Integer.parseInt(coords[1])));
+                        map.addTeleportTo(new Point(Integer.parseInt(coords[4]), Integer.parseInt(coords[5])));
+                        // Here coords[6] is the orientation, no element in settings for this, can be added if needed
+                        break;
+                    case "texture":
+                        map.addTexture(new Rectangle2D.Double(Double.parseDouble(coords[0]), Double.parseDouble(coords[1]), Integer.parseInt(coords[2]) - Integer.parseInt(coords[0]), Integer.parseInt(coords[3]) - Integer.parseInt(coords[1])));
+                        map.addTextureType(Integer.parseInt(coords[4]));
+                        // Here coords[5] is the orientation, can also be added like above for portals
+                        break;
+                    case "targetArea":
+                        map.setTargetArea(new Rectangle2D.Double(Double.parseDouble(coords[0]), Double.parseDouble(coords[1]), Integer.parseInt(coords[2]) - Integer.parseInt(coords[0]), Integer.parseInt(coords[3]) - Integer.parseInt(coords[1])));
+                        break;
+                    case "spawnAreaIntruders":
+                        map.setSpawnAreaIntruders(new Rectangle2D.Double(Double.parseDouble(coords[0]), Double.parseDouble(coords[1]), Integer.parseInt(coords[2]) - Integer.parseInt(coords[0]), Integer.parseInt(coords[3]) - Integer.parseInt(coords[1])));
+                        break;
+                    case "spawnAreaGuards":
+                        map.setSpawnAreaGuards(new Rectangle2D.Double(Double.parseDouble(coords[0]), Double.parseDouble(coords[1]), Integer.parseInt(coords[2]) - Integer.parseInt(coords[0]), Integer.parseInt(coords[3]) - Integer.parseInt(coords[1])));
```

Comment:
Thanks, pushed those changes. Doesn't look as nice as the example you gave, because when tabbing to put the parameters at the same level of indentation, it still uses over 100 columns. Any other improvements let me know.